Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,293 @@
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest(Assert, @"{""cid"":3,""dataRate"":1,""txPower"":2,""chMask"":3,""chMaskCntl"":4,""nbRep"":5}");
+
+            [Fact]
+            public void FromBytes_Success() => FromBytesTest(Assert, bytes => new LinkADRRequest(bytes), DataBytes);
+        }
+
+        public sealed class LinkCheckAnswerTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.LinkCheckCmd;
+            public override LinkCheckAnswer Subject => new LinkCheckAnswer(1, 2);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 1, 2 };
+            public override int Length => 3;
+
+            [Fact]
+            public void FromBytes_Success() => FromBytesTest(actual => actual.Cid == Cid
+                                                                    && actual.Margin == Subject.Margin
+                                                                    && actual.GwCnt == Subject.GwCnt,
+                                                             bytes => new LinkCheckAnswer(new ReadOnlySpan<byte>(bytes)),
+                                                             DataBytes);
+        }
+
+        public sealed class LinkCheckRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.LinkCheckCmd;
+            public override MacCommand Subject => new LinkCheckRequest();
+            public override IReadOnlyList<byte> DataBytes => Array.Empty<byte>();
+            public override int Length => 1;
+        }
+
+        public sealed class NewChannelAnswerTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.NewChannelCmd;
+            public override NewChannelAnswer Subject => new NewChannelAnswer(false, true);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 1 };
+            public override int Length => 2;
+
+            [Fact]
+            public void FromBytes_Success() => FromBytesTest(actual => actual.Cid == Cid
+                                                                    && !actual.DataRangeOk
+                                                                    && actual.ChannelFreqOk,
+                                                             bytes => new NewChannelAnswer(new ReadOnlySpan<byte>(bytes)),
+                                                             DataBytes);
+        }
+
+        public sealed class NewChannelRequestTests : MacCommandTests
+        {
+            public override Cid Cid => Cid.NewChannelCmd;
+            public override NewChannelRequest Subject => new NewChannelRequest(1, 2, 3, 4);
+            public override IReadOnlyList<byte> DataBytes => new byte[] { 1, 2, 0, 0, 0b110100 };
+            public override int Length => 6;
+
+            [Fact]
+            public void Deserializes_Correctly() =>
+                DeserializationTest<NewChannelRequest>(actual => actual.Cid == Cid
```

Comment:
There might be a trick to avoid these equality tests by simply returning an anonymous object given a command. Since anonymous types implement equality, you could then use a regular `Assert.Equal` to compare the subject and the actual value. It would also have the benefit of more helpful assertion failure messages than `Assert.True` does.

Reply:
Nice trick, applied with 78ee5aa5d208c6bbe74ab064f6ae84de1a6c3e20